src/file_handler.rs

```diff
@@ -123,22 +124,94 @@ fn config_dir_path() -> String {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use file_handler::ConfigManagement;
+    use file_handler::FileManagement;
     use file_handler::FileSystem;
+    use std::fs;
     use std::io;
 
     struct MockFileSystem;
+    struct MockFileHandler;
 
     impl FileSystem for MockFileSystem {
         fn create_dir(&self, _path: &str) -> io::Result<()> {
             Ok(())
         }
     }
 
+    impl ConfigManagement for MockFileHandler {
+        fn config_dir_create(&self) -> io::Result<String> {
+            Ok(config_dir_path())
+        }
+
+        fn config_dir_exists(&self) -> bool {
+            self.file_exists(&config_dir_path())
+        }
+
+        fn config_read(&self, _file: ConfigFile) -> io::Result<String> {
+            let contents = String::from("~/idea");
+            Ok(contents)
+        }
+
+        fn config_write(&self, _file: ConfigFile, _value: String) -> io::Result<()> {
+            Ok(())
+        }
+    }
+
+    impl FileManagement for MockFileHandler {
+        fn file_exists(&self, path: &str) -> bool {
+            fs::metadata(path).is_ok()
```

Comment:
So this would also have to be replaced by something else (same reason as above). For now we can just return `true`.